src/client/components/facet_results/ResultTable.js

```diff
@@ -99,8 +99,8 @@ class ResultTable extends React.Component {
       search: `?page=${page}`
     })
 
-    // check if facet updates have been made before
-    if (this.props.facetUpdateID > 0) {
+    // check if facet updates have been made before or re-fetch the result when the language change
+    if (this.props.facetUpdateID > 0 || this.props.currentLocale) {
```

Comment:
This is in componentDidMount which is only called on mount. The currentLocale check should probably be in componentDidUpdate.

Reply:
Hi @evsheino, I know the logic should be in componentDidUpdate, but I don't know why `prevProps.currentLocale` is always equal to `this.props.currentLocale`. This is a quick fix before I found the better solution to make it work in componentDidUpdate.